perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/GroupsManagerBlImpl.java

```diff
@@ -3762,10 +3765,63 @@ public void updateExistingMemberWhileSynchronization(PerunSession sess, Group gr
 		//Synchronize userExtSources (add not existing)
 		addUserExtSources(sess, candidate, memberToUpdate);
 
+		synchronizeGroupStatus(sess, memberToUpdate, candidate.getExpectedSyncGroupStatus(), group);
```

Comment:
This will still get called before the member gets added to the group in some cases - from `addMissingMemberWhileSynchronization` (and will fail).